src/_imaging.c

```diff
@@ -215,16 +198,12 @@ PyImaging_AsImaging(PyObject *op) {
 
 void
 ImagingSectionEnter(ImagingSectionCookie *cookie) {
-#ifdef WITH_THREADING
```

Comment:
I've tried to measure performance penalty for saving state.

Empty function call:

```python
In [2]: %timeit Image.core.test_threading()
64.1 ns ± 0.842 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

With `ImagingSectionEnter`/`ImagingSectionLeave`:
```python
In [6]: %timeit Image.core.test_threading()
109 ns ± 0.951 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```